src/kube-runtime/src/runtime

```diff
@@ -55,12 +55,15 @@ ${PAI_RUNTIME_DIR}/precommands.sh > >(tee -a ${PAI_LOG_DIR}/runtime.pai.stdout)
 # Put verbose output to user.pai.all, stdout to user.pai.stdout, stderr to user.pai.stderr
 # execute user commands
 # priority=100
-echo "[INFO] USER COMMAND START"
+echo "[INFO] USER COMMAND START" >> ${PAI_LOG_DIR}/user.pai.stderr
 ${PAI_RUNTIME_DIR}/user.sh > >(tee -a ${PAI_LOG_DIR}/user.pai.stdout ${PAI_USER_LOG_FILE}) 2> >(tee -a ${PAI_LOG_DIR}/user.pai.stderr ${PAI_USER_LOG_FILE} >&2) &
 USER_PID=$!
 
-echo "[INFO] USER COMMAND END"
 wait ${USER_PID}
+USER_EXIT_CODE=$?
+echo "[INFO] USER COMMAND END" >> ${PAI_LOG_DIR}/user.pai.stderr
 
 # execute postCommands generated by plugin
-${PAI_RUNTIME_DIR}/postcommands.sh > >(tee -a ${PAI_LOG_DIR}/runtime.pai.stdout) 2> >(tee -a ${PAI_LOG_DIR}/runtime.pai.stderr >&2)
\ No newline at end of file
+${PAI_RUNTIME_DIR}/postcommands.sh > >(tee -a ${PAI_LOG_DIR}/runtime.pai.stdout) 2> >(tee -a ${PAI_LOG_DIR}/runtime.pai.stderr >&2)
+echo "container will exit with code ${USER_EXIT_CODE}" >> ${PAI_LOG_DIR}/runtime.pai.stderr
+exit ${USER_EXIT_CODE}
```

Comment:
what if `precommands.sh` or `postcommands.sh` failed? better to add `|| exit $?` at each line's ending

Reply:
That likes "set -o errexit", so I suggest add them all in header.

Reply:
Dian create another PR for this: https://github.com/microsoft/pai/pull/3559, will close this PR